src/lib/svelte-query/queries/metric-view.ts

```diff
@@ -181,6 +182,15 @@ export const getMetricViewTotalsQueryKey = (
   return [TotalsId, metricsDefId, isReferenceValue];
 };
 
+export const invalidateMetricView = async (
+  queryClient: QueryClient,
+  metricsDefId: string
+) => {
+  // wait for meta to be invalidated
+  await queryClient.invalidateQueries([MetaId, metricsDefId]);
```

Comment:
Does this `await` wait for the refetch to complete? If so, then we're good. If not, then I wonder if we'll have a race condition when `invalidateMetricViewData` runs.

Reply:
This seems to be the case after checking with breakpoints.